public/v2/apps/discourse.json

```diff
@@ -0,0 +1,177 @@
+                    "FROM docker.io/bitnami/discourse:$$cap_discourse_version",
+                    "CMD [\"nami\" , \"start\" , \"--foreground\" , \"discourse-sidekiq\"]"
+                ],
+                "environment": {
+                    "DISCOURSE_POSTGRESQL_NAME": "$$cap_db_name",
+                    "DISCOURSE_POSTGRESQL_USERNAME": "$$cap_db_user",
+                    "DISCOURSE_POSTGRESQL_PASSWORD": "$$cap_db_pass",
+                    "DISCOURSE_HOST": "srv-captain--$$cap_appname",
+                    "DISCOURSE_PORT": "3000",
+                    "DISCOURSE_HOSTNAME": "$$cap_appname.$$cap_root_domain",
+                    "POSTGRESQL_HOST": "srv-captain--$$cap_appname-postgresql",
+                    "REDIS_HOST": "srv-captain--$$cap_appname-redis",
+                    "SMTP_HOST": "$$cap_smtp_host",
+                    "SMTP_PORT": "$$cap_smtp_port",
+                    "SMTP_USER": "$$cap_smtp_user",
+                    "SMTP_PASSWORD": "$$cap_smtp_pass"
+                }
+            }
+        }
+    },
+    "instructions": {
+        "start": "Discourse is the 100% open source discussion platform built for the next decade of the Internet. Use it as a mailing list, discussion forum, long-form chat room, and more!",
+        "end": "Discourse is deployed and will be available in the next few minutes as $$cap_appname"
+    },
+    "variables": [
+        {
+            "id": "$$cap_discourse_version",
+            "label": "Discourse Version",
+            "defaultValue": "2-debian-10",
+            "description": "Version of Discourse",
+            "validRegex": "/^([^\\s^\\/])+$/"
+        },
+        {
+            "id": "$$cap_postgresql_version",
+            "label": "Postgresql (database) version",
+            "defaultValue": "11-debian-10",
+            "description": "Version of Postgresql",
+            "validRegex": "/^([^\\s^\\/])+$/"
+        },
+        {
+            "id": "$$cap_sitename",
+            "label": "Sitename",
+            "defaultValue": "My site!",
+            "description": "The title of the site, the words on the tab"
+        },
+        {
+            "id": "$$cap_redis_version",
+            "label": "Redis version",
+            "defaultValue": "5.0-debian-10",
+            "description": "Version of Redis",
+            "validRegex": "/^([^\\s^\\/])+$/"
+        },
+        {
+            "id": "$$cap_admin_user",
+            "label": "Admin username",
+            "defaultValue": "admin",
+            "description": "Name of the admin account"
+        },
+        {
+            "id": "$$cap_admin_pass",
+            "label": "Admin password",
+            "defaultValue": "sTr0nG_p@s$wo%d",
+            "description": "Password of the admin account"
+        },
+        {
+            "id": "$$cap_admin_email",
+            "label": "Admin email",
+            "defaultValue": "your_email@gmail.com",
+            "description": "Email of the admin account"
```

Comment:
Let's add regex to make sure that user, password, and email are filled. It can be as simple as `"validRegex": "/.{1,}/"`

Reply:
Okay.